ion/reader.go

```diff
@@ -292,62 +295,66 @@ func (r *reader) IntSize() (IntSize, error) {
 }
 
 // IntValue returns the current value as an int.
-func (r *reader) IntValue() (int, error) {
+func (r *reader) IntValue() (*int, error) {
 	i, err := r.Int64Value()
```

Comment:
Should we have an if-statement here to check the type and throw an appropriate error if the function is called on an incorrect Ion Type?

Reply:
```r.Int64Value()``` here calls another function in this file (right below) [```Int64Value```](https://github.com/amzn/ion-go/pull/157/files#diff-0a0ff18f6967f2ebfe8b753560b3c836R311) where the check on type is done.

Reply:
You are right. I missed that, thanks.